aptos/token_bridge/sources/bridge_state.move

```diff
@@ -118,13 +148,127 @@ module token_bridge::bridge_state {
+        set_wrapped_asset(&native_info, token_address);
+
+        // store coin capabilities
+        let _token_bridge_signer = token_bridge_signer();
+        let coin_caps = CoinCapabilities<CoinType> { mint_cap: mint_cap, freeze_cap: freeze_cap, burn_cap: burn_cap};
+        move_to(&_token_bridge_signer, coin_caps);
+
+        vaa::destroy(vaa);
+    }
+
+    // transfer a native or wraped token from sender to token_bridge
+    public entry fun transfer_tokens<CoinType>(coins: Coin<CoinType>, relayer_fee: u128): TransferResult acquires State {
+        assert!(coin::is_account_registered<CoinType>(@token_bridge), E_COIN_NOT_REGISTERED);
+
+        let token_address = hash_type_info<CoinType>();
+        let amount = coin::value<CoinType>(&coins);
+
+        // transfer tokens from sender to token_bridge
+        coin::deposit<CoinType>(@token_bridge, coins);
+
+        // return TransferResult encapsulating details of token transferred
+        let origin_chain;
+        let origin_address;
+
+        if (is_wrapped_asset(token_address)) {
+            let _native_info = native_info(token_address);
+            origin_chain = _native_info.token_chain;
+            origin_address = _native_info.token_address;
+        } else {
+             if (!is_registered_native_asset(token_address)) {
+                set_native_asset(token_address, type_of<CoinType>());
+             };
+            origin_chain = get_chain_id();
+            origin_address = token_address;
+        };
+
+        // TODO - normalize amount by using helpers from utils.move
+        let normalized_amount = u256::from_u64(amount);
+        // TODO - normalize relayer fee
+        let normalized_relayer_fee = u256::from_u128(relayer_fee);
+        let wormhole_fee = u256::from_u64(0);
+
+        let transfer_result: TransferResult = create_transfer_result(
+            origin_chain,
+            origin_address,
+            normalized_amount,
+            normalized_relayer_fee,
+            wormhole_fee
+        );
+        transfer_result
+    }
+
+    public entry fun transfer_tokens_with_signer<CoinType>(sender: &signer, amount: u64, relayer_fee: u128): TransferResult acquires State {
+        let coins = coin::withdraw<CoinType>(sender, amount);
+        transfer_tokens<CoinType>(coins, relayer_fee)
+    }
+
+    public fun log_transfer(
+        token_chain: U16,
+        token_address: vector<u8>,
+        amount: U256,
+        recipient_chain: U16,
+        recipient: vector<u8>,
+        fee: U256,
+        message_fee: Coin<AptosCoin>,
+        nonce: u64
+    ): u64 acquires State{
+        let fee_value = coin::value<AptosCoin>(&message_fee);
+        assert!(u256::compare(&u256::from_u64(fee_value), &amount)==1, E_FEE_EXCEEDS_AMOUNT);
+        // TODO - payloadID is 1 for token transfer?
+        let transfer = bridge_structs::create_transfer(1, amount, token_address, token_chain, recipient, recipient_chain, fee);
```

Comment:
create_transfer should not take the payload id as an argument, and the payload id should not be part of the struct. but we can fix that later (I think this is a "bug" in the EVM contracts, at least from a coding style perspective)

Reply:
ok